# What is business risk & how to measure it in python?

__One of the key aspects to do well in the stock market is to know what is business risk.__ <br>Business risk represents external and internal factors that may lead a company to a reduction of profits or even bankruptcy.<br><br>Companies should have a risk management plan in order to minimise and mitigate risk. This is part of the business risk management process.

<br>

We could directly measure business risk by using standard deviation on recent company financials. For instance, we could take company’s financial indicators. Such as operating income, revenue or net income. Then, we could calculate the standard deviation of these metrics to analyse the variation in recent years.<br><br> __The lower the standard deviation, the lower the variation. A lower variation means that a business have less uncertainty about future firm performance.__

We will adjust for the size of the company to get rid of this comparability issue. This way we can represent the standard deviation of key metrics in a format enabling comparison across firms.

To standardise above mentioned business risk metrics and enable comparison across firms of different sizes, we can use the coefficient of variation. The coefficient of variation is computed by dividing standard deviation by the mean.

# Below is an example on how can we calculate the coefficient of variation for sales:

__Revenues Coefficient of Variation = Standard Deviation Revenues / Revenues Mean__

By using the coefficient of variation instead of the standard deviation, we can compare business risk across firms of different sizes.

<br>

# Calculating business risk in python

To keep things simple, we will only calculate business risk based on sales, operating income and net income. And we will use the coefficient of variation for that purpose.

We will start with a list of 3 stocks; GAIL, ONGC and INDIAN OIL CORP. Then we will compare business risk across them. Feel free to add more companies to the analysis.

First, we will use an API, financialmodelingprep. It offers around 200 free API calls per day upon registration. With this API, we can extract company revenues, operating income and net income for the last years.

To extract each company metrics, we simply pass the company ticker to the url as per below. Note that you need to store your API key into the demo variable.

Then, we append the extracted values into a list. Each of the list will contain the latest 5 year company revenues, operating income and net income.

<br><br>

After having revenues, operating income and net income from the last 5 years, all included in Python lists, we need to convert them into Numpy arrays. We do this since with Numpy is super easy to calculate the mean and standard deviation:

Finally, we can simply apply our coefficient of variation formula to each of the metrics

In [19]:
import requests

demo = '7af0198f40a084eacb20ebf1e07a71d8'
stocks = ['AMZN', 'NFLX','NVDA']

for stock in stocks:
  income_statement = requests.get(f'https://financialmodelingprep.com/api/v3/income-statement/{stock}?apikey={demo}').json()

  number_of_years = 0

  revenues = []
  operating_income = []
  net_income = []

  for item in income_statement:
    if number_of_years < 5:
      revenues.append(income_statement[number_of_years]['revenue'])
      operating_income.append(income_statement[number_of_years]['operatingIncome'])
      net_income.append(income_statement[number_of_years]['netIncome'])
      number_of_years += 1

  
#import numpy to convert list to array:
  import numpy
  revenues_array = numpy.array(revenues)
  operating_income_array = numpy.array(operating_income)
  net_income_array = numpy.array(net_income)
  CV_Sales = revenues_array.std() / revenues_array.mean()
  print('Revenue Coefficient of Variation for '+ stock + ' is ' + str(round(CV_Sales,2)))

  CV_OI = operating_income_array.std() / operating_income_array.mean()
  print('Operating Income Coefficient of Variation for '+ stock + ' is ' + str(round(CV_OI,2)))

  CV_Net_Income = net_income_array.std() / net_income_array.mean()
  print('Net Income Coefficient of Variation for '+ stock + ' is ' + str(round(CV_Net_Income,2))+'\n')


Revenue Coefficient of Variation for AMZN is 0.36
Operating Income Coefficient of Variation for AMZN is 0.61
Net Income Coefficient of Variation for AMZN is 0.71

Revenue Coefficient of Variation for NFLX is 0.36
Operating Income Coefficient of Variation for NFLX is 0.75
Net Income Coefficient of Variation for NFLX is 0.7

Revenue Coefficient of Variation for NVDA is 0.29
Operating Income Coefficient of Variation for NVDA is 0.27
Net Income Coefficient of Variation for NVDA is 0.3



By taking revenue, operating income and net income as proxy to measure business risk, and based on the last 5 years data, we can see that NVIDIA is by far the company with the lower business risk. It has a variation of around 0.29.

On the other side, Amazon and Netflix have a comparatively higher business risk with a sales variation coefficient of 0.36. The other two metrics, operating income and net income, seems to support this finding as well.

__Important note:__ Past variation may not always be a good indication for future variation. There may be significant and unexpected events happening in the future which may impact company financials.